lib/tests/formulae_dependents.rb

```diff
@@ -171,17 +171,18 @@ def install_dependent(dependent, testable_dependents, args:, build_from_source:
         if testable_dependents.include? dependent
           test "brew", "install", "--only-dependencies", "--include-test", dependent.full_name
 
-          dependent.recursive_dependencies.each do |dependency|
-            next if dependency.build? && !dependency.test?
-
+          # Traverse the dependency tree to check for formulae we need to link
+          dependencies_to_link = dependent.recursive_dependencies do |dep_dependent, dependency|
```

Comment:
Internal caching is useful for very large trees:

```suggestion
          dependencies_to_link = Dependency.expand(dependent, cache_key: "test-bot-link-#{dependent.full_name}-#{Time.now.to_f}") do |dep_dependent, dependency|
```

We however don't want to cache it between calls since the state of `.linked_keg.exist?` may change (hence the `Time.now`). That could probably also be simplified to `.linked?` I think?

Reply:
Do we really want the cache to be invalidated every microsecond?